vagrant/ansible/roles/sit.cephfs/tasks/main.yml

```diff
@@ -0,0 +1,77 @@
+---
+- name: Ensure /etc/ceph exists
+  file:
+    path: /etc/ceph
+    state: directory
+
+- name: Install Ceph
+  delegate_to: "{{ config.groups['cluster'][0] }}"
+  run_once: true
+  block:
+    - name: Bootstrap the ceph node
+      command: >
+        /root/cephadm bootstrap
+        --mon-ip {{ config.nodes[inventory_hostname].networks.private }}
+        --ssh-private-key /root/.ssh/id_rsa
+        --ssh-public-key /root/.ssh/id_rsa.pub
+        --skip-monitoring-stack
+      args:
+        creates: /etc/ceph/ceph.conf
+
+    - name: Add other ceph nodes
+      command: /root/cephadm shell -- ceph orch host add "{{ item }}" --labels _admin
+      loop: "{{ config.groups['cluster'][1:] }}"
+
+    - name: Set mon count to 3
+      command: /root/cephadm shell -- ceph orch apply mon 3
+
+    - name: Add all available devices
+      command: /root/cephadm shell -- ceph orch apply osd --all-available-devices
+
+    - name: Add an MDS
+      command: /root/cephadm shell -- ceph orch apply mds cephfs --placement=1
+
+    - name: Create a CephFS instance
+      command: /root/cephadm shell -- ceph fs volume create sit_fs
```

Comment:
If a heavy load causes an MDS to crash, shouldn't we detect it and fix it ? the purpose of this test environment is to detect problems. Adding standby MDS's in case of failure wouldn't hide those problems ?

If everything goes as expected, the standby MDS's won't be used, but they will use valuable resources (note that everything works in VMs in the same physical machine). On the other hand, if it fails, I would prefer to detect it.

Reply:
You can get core dumps, back traces in MDS logs or health warnings (logged in cluster logs) for any crashes/failures  with/without standby daemons; we are not suppressing anything. I'm not sure about what workloads you guys have but there are many cases in our test suite where there are intentional MDS failure and takeovers(this also makes sure standbys do their job). This also limits testing a lot of cases, one such example could be let's say you want to create a CephFS for testing then because there are no standbys and the active MDS rank is assigned to the fs `sit_fs` you'd get a health warning:
```
HEALTH_WARN insufficient standby MDS daemons available
[WRN] MDS_INSUFFICIENT_STANDBY: insufficient standby MDS daemons available
have 0; want 1 more
```

I'd say having standbys won't suppress anything but make the testing env more robust and capable of handling all  the cases.

Reply:
If a crash in the MDS happens and there's a standby MDS, is it possible that the client doesn't see any failure ? right now the testing is done just from the samba side, and the only failures reported are those that return unexpected errors/data. If a crash can be hidden, then we'll also need to monitor things outside samba tests to know about the problem.

Is that right ?

Reply:
client reconnects to the failed over MDS(one of the standbys), it can also retry some requests when the MDS is in `up:clientreplay` state. Also in cases where the requests couldn't be journal-ed or the journal isn't flushed then the clients can lose some of it's requests as well. So the client knows when MDS failed but If you say your workload is such that you want/expect a complete halt of everything then maybe a single MDS might be the suitable thing to do.

Reply:
The environment is able to use different backends, but the tests only check samba functionality (at least for now). A failure in a test means a bug in samba or a bug in the backend, but in no case any of the tests will be aware of which backend is being used, so there won't be any attempt to voluntarily cause a failure of the backend. If any failure happens in the backend, we prefer to have a clear failure in the test. So, if adding standby MDS can hide some of these failures, I think it's not what we intend to do right now.

Reply:
I think the failures would be visible but I respect your decision if you feel safe this way